File: src/main/java/com/mbayou/kick4k/moderation/ModerationClient.java
package com.mbayou.kick4k.moderation;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.mbayou.kick4k.KickConfiguration;
import com.mbayou.kick4k.api.ApiClient;
import com.mbayou.kick4k.authorization.AuthorizationClient;

import java.net.http.HttpClient;

public class ModerationClient extends ApiClient {
    public ModerationClient(HttpClient httpClient, ObjectMapper mapper, KickConfiguration configuration, AuthorizationClient authorization) {
        super(httpClient, mapper, configuration, authorization);
    }

    public void postModerationBans(PostModerationBansRequest request) {
        this.post(this.configuration.getModeration())
                .body(request)
                .send(new TypeReference<>() {});
    }

    public void deleteModerationBans(DeleteModerationBansRequest request) {
        this.delete(this.configuration.getModeration())
                .body(request)
                .send(new TypeReference<>() {});
    }

    public void deleteModerationBans(Integer broadcasterUserId, Integer userId) {
        this.deleteModerationBans(new DeleteModerationBansRequest(broadcasterUserId, userId));
    }
}

File: src/main/kotlin/com/mbayou/kick4k/moderation/ModerationClient.kt
package com.mbayou.kick4k.moderation

import com.fasterxml.jackson.databind.ObjectMapper
import com.mbayou.kick4k.KickConfiguration
import com.mbayou.kick4k.api.ApiClient
import java.net.http.HttpClient

class ModerationClient(
    httpClient: HttpClient,
    mapper: ObjectMapper,
    configuration: KickConfiguration,
): ApiClient(httpClient, mapper, configuration) {

    fun postModerationBans(accessToken: String, request: PostModerationBansRequest) {
        post(configuration.moderation)
            .body(request)
            .withAccessToken(accessToken)
            .send(responseType<Unit>())
    }

    fun deleteModerationBans(accessToken: String, request: DeleteModerationBansRequest) {
        delete(configuration.moderation)
            .body(request)
            .withAccessToken(accessToken)
            .send(responseType<Unit>())
    }

    fun deleteModerationBans(accessToken: String, broadcasterUserId: Int, userId: Int) {
        deleteModerationBans(accessToken, DeleteModerationBansRequest(broadcasterUserId, userId))
    }
}
